### DISCUSSION TITLE
select_nodes

### DISCUSSION THREAD

**Discussion by jtaylorme** [zeux/pugixml#569] [Q&A] [answered]
First, this is a VERY nice library.  I appreciate all the work you have done.

Second,

     Is there a way to use select_nodes starting from a specific node?  I can select them from the document but when I try from a node it never returns results even though there is a method for it.

     If I search from the root for "/CATALOG/CD" in my XML it works correctly.   If I search for "/CATALOG" and then use that node as the starting point and search for "CD" I get nothing.   

    Am I misunderstanding the point of node.select_nodes()?  Perhaps it is an XPath nuance?

   Thanks.   

jim

**Comment by zeux** [comment, marked as answer]
This works for me as expected:

```c++
#include "pugixml.hpp"
#include <stdio.h>

int main(int argc, char **argv)
{
   pugi::xml_document doc;
   doc.load_string("<CATALOG><CD>1</CD><CD>2</CD></CATALOG>");

   pugi::xml_node catalog = doc.select_node("/CATALOG").node();

   for (pugi::xpath_node cd : catalog.select_nodes("CD"))
   {
           printf("CD text %s\n", cd.node().text().get());
   }

   return 0;
}
```

**Comment by jtaylorme** [comment]
Sorry to have bothered you.   I see where I went wrong.   After an hour or two of trying different things I decided to ask but turns out it was a incorrect vector reference.  Thanks.

Jim
